### PR TITLE
Rename eps -> ulp

### PR DESCRIPTION
**Pull request by Keno**
As discussed on triage, this renames `eps` to `ulp`, because the former is a bit of a pun and even the docstring says this computes the `unit in last place` of the given value. Particularly, because we extend this function to things like Dates, `eps` is a bit of an awkward name.

### CODE REVIEW

**Review by ararslan — APPROVED** [review]

File: base/float.jl [hunk 1]
```diff
@@ -767,74 +767,74 @@ realmin() = realmin(Float64)
 realmax() = realmax(Float64)
 
 """
-    eps(::Type{T}) where T<:AbstractFloat
-    eps()
+    ulp(::Type{T}) where T<:AbstractFloat
+    ulp()
 
 Return the *machine epsilon* of the floating point type `T` (`T = Float64` by
```
Comment (waldyrious) [inline base/float.jl]: Shouldn't "machine epsilon" be updated (or adjusted to also mention ULP) here as well?

**Comment by stevengj** [comment]
`eps` or `epsilon` is pretty universal here: Matlab, Numpy, R, C++, Fortran.   It seems crazy to me to abandon it in the 11th hour on the vague grounds that it is "a bit of a pun".

I also don't see any problem in using it for dates or other types — it seems just as easy to define `eps` as it is to define `ulp` for those types, since the term "ulp" is also rarely if ever used for [anything but floating-point arithmetic](https://en.wikipedia.org/wiki/Unit_in_the_last_place).

**Comment by stevengj** [comment]
I don't like the term "sine" — it's a bad Latin pun arising from an [Arabic translation error](https://en.wikipedia.org/wiki/Sine), and "sin" is even worse.   I propose that we rename it the function to `hac` (for "half a chord"), which is more logical and agrees with the original Sanskrit.   Let's just sneak in the rename right before the 0.7 release, I'm sure no one will mind.

Or we could continue to call this function what everyone else calls it.

**Comment by KristofferC** [comment]
Or `jiba`?

**Comment by ViralBShah** [comment]
I knew `eps`/`epsilon` was generally used, but didn't realize it was this universal.

**Comment by JeffBezanson** [comment]
Same here; if it's that widely used let's just close this.

**Pull request closed without merging**